src/dextrader/analysis/arbitrage.py

```diff
@@ -0,0 +1,109 @@
+                               df2: pd.DataFrame,
+                               pair: str,
+                               mkt1: str,
+                               mkt2: str,
+                               threshold: float = 0.8) -> dict:
+    '''
+    :param df1: dataframe of market A data from nomics candles api data for timeframe X, for given pair
+    :param df2: dataframe of market B data from nomics candles api data from timeframe X, for given pair
+    :param pair: string identifier for pair that data represents
+    :param mkt1: string identifier for pair market 1
+    :param mkt2: string identifier for pair market 2
+    :param threshold: threshold for considering trades profitable
+    :return: arbitrage_results (dict): in the form:
+                { 'info' : [pairstring, market_1, market_2],
+                  'dataframe' : <pd.DataFrame>,
+                  'profitable_trades_min' : int
+                  'profitable_trades_close' : int
+                }
+    '''
+
+    # Merge data with shared timestamps
+    merged_data = pd.merge(df1, df2, how='outer', on='Datetime')
+
+    # Convert data to numeric data
+    cols = ['low_x', 'low_y', 'open_x', 'open_y', 'close_x', 'close_y', 'high_x', 'high_y']
+    merged_data[cols] = merged_data[cols].apply(pd.to_numeric, errors='coerce')
+
+    # arbitrage
+    merged_data['arbitrage'] = (merged_data['close_x'] - merged_data['close_y']).abs()
+
+    # Compute arbitrage from close values
+    merged_data = merged_data.assign(arbitrage_close=
+                                     lambda x: np.where(x['close_x'] > x['close_y'],
+                                                        (x['close_x'] - x['close_y'])/x['close_x'],
+                                                        (x['close_y'] - x['close_x'])/x['close_y']))
+    merged_data['arbitrage_close'] *= 100
+
+    # Compute min arbitrage (worst case)
+    merged_data = merged_data.assign(arbitrage_min=
+                                     lambda x: np.where(x['open_x'] > x['open_y'],
+                                                        (x['low_x'] - x['high_y'])/x['open_y'],
+                                                        (x['low_y'] - x['high_x'])/x['open_x']))
+    merged_data['arbitrage_min'] *= 100
+
+    # Find trades that exceed the profit threshold
+    merged_data = merged_data.assign(profitable_trades_close=
+                                     lambda x: x['arbitrage_close'] > threshold)
+    merged_data = merged_data.assign(profitable_trades_min=
+                                     lambda x: x['arbitrage_min'] > threshold)
+
+    # Get time in int64
+    merged_data['datetime64'] = merged_data['Datetime'].astype('datetime64[ns]')
+
+    arbitrage_results = dict()
+    arbitrage_results['info'] = (pair, mkt1, mkt2)
+    arbitrage_results['dataframe'] = merged_data
+    arbitrage_results['profitable_trades_min'] = np.sum(merged_data['profitable_trades_close'].to_numpy().astype(int))
```

Comment:
Lines 64-65, both sum `profitable_trades_close`. Is this one supposed to be `profitable_trades_min` ?

Reply:
nice catch